app/latigo/time_series_api/prediction_storage_provider.py

```diff
@@ -61,11 +63,21 @@ def put_prediction(self, prediction_data: PredictionDataSet):
                 continue
             output_time_series_ids[col] = ""
             description = OutputTag.make_output_tag_description(operation, tag_name)
-            meta, err = self._create_id_if_not_exists(
-                name=output_tag_name,
-                description=description,
-                asset_id=common_asset_id,
-            )
+            try:
+                meta, err = self._create_id_if_not_exists(
+                    name=output_tag_name,
+                    description=description,
+                    asset_id=common_asset_id,
+                )
+            except HTTPError as error:
+                if error.response.status_code == 409:
+                    # if such tag_name might already exists in the TS try to get/create once more.
+                    meta, err = self.replace_cached_metadata_with_new(
+                        tag_name=output_tag_name, asset_id=common_asset_id, description=description
+                    )
+                else:
+                    raise
```

Comment:
```suggestion
                if error.response.status_code != 409:
                    raise
                    
                # if such tag_name might already exists in the TS try to get/create once more.
                meta, err = self.replace_cached_metadata_with_new(
                    tag_name=output_tag_name, asset_id=common_asset_id, description=description
                )
```
Having a short-circuit break improves readability.

Reply:
good point, thanks